apps/codeimage/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,47 @@
+import {Box, IconButton} from '@codeimage/ui';
+import {createEffect, For, mergeProps, ParentProps, Setter} from 'solid-js';
+import {usePaginationButtons} from './buttons';
+import Page from './page';
+
+export interface PaginationProps {
+  /**
+   * The current value (controlled).
+   */
+  value?: number;
+  /**
+   * Handler that is called when the value changes.
+   */
+  onChange: Setter<number>;
+  maxValue?: number;
```

Comment:
Maybe `value` and `maxValue` property names could be changed to `pageNumber` and `pageSize` 🤔